Libraries/PeriphDrivers/Source/LP/lp_me14.c

```diff
@@ -574,3 +549,226 @@ void MXC_LP_ICacheXIPLightSleepEnable(void)
+    } else {
+        /* ICC power up */
+        MXC_PWRSEQ->lpmemsd &= ~MXC_F_PWRSEQ_LPMEMSD_IC1SD;
+        /* Enable */
+        if (icc1_state & (1 << CTRL_POS))
+            MXC_ICC_RevA_Enable((mxc_icc_reva_regs_t *)MXC_ICC1);
+    }
+
+    /* Restore ICC0 state */
+    if (icc0_state & (1 << ICSD_POS)) {
+        /* ICC power down. Do not restore Enable state. */
+        MXC_PWRSEQ->lpmemsd |= MXC_F_PWRSEQ_LPMEMSD_ICACHESD;
+    } else {
+        /* ICC power up */
+        MXC_PWRSEQ->lpmemsd &= ~MXC_F_PWRSEQ_LPMEMSD_ICACHESD;
+        /* Enable */
+        if (icc0_state & (1 << CTRL_POS))
+            MXC_ICC_RevA_Enable((mxc_icc_reva_regs_t *)MXC_ICC0);
+    }
+}
+
+void MXC_LP_EnterDeepSleepMode(void)
+{
+    save_preDeepSleep_state();
+
+    /*void prepForDeepSleep(void)*/
+    {
+        MXC_ICC_Disable();
+        MXC_LP_ICache0Shutdown();
+
+        /* Shutdown unused power domains */
+        MXC_PWRSEQ->lpcn |= MXC_F_PWRSEQ_LPCN_BGOFF;
+
+        switchToHIRCD4();
+
+        /* Set SIMO clock, VDDCSW, VregO_B voltage for DeepSleep */
+        /*void MXC_LP_SIMOprepForDeepSleep(uint32_t voltage)*/
+        {
+            /* Prevent SIMO soft start on wakeup */
+            MXC_LP_FastWakeupDisable();
+
+            /* Enable VDDCSWEN=1 prior to enter DEEPSLEEP */
+            MXC_MCR->ctrl |= MXC_F_MCR_CTRL_VDDCSWEN;
+
+            /* SIMO clock setup for deep sleep */
+            *(volatile int *)0x40005434 = 1; /* SIMOCLKDIV [1:0] : 0=div1; 1=div8; 2=div1 3=div16 */
+            /* BUCK_CLKSEL [25:24] : 0=8K; 1=16K; 2=30K; 3=RFU */
+            *(volatile int *)0x40005440 = (*(volatile int *)0x40005440 & (~(0x3 << 24))) |
+                                          (0x2 << 24);
+            /* BUCK_CLKSEL_LP [7:6] : 0=8K; 1=16K; 2=30K; 3=RFU */
+            *(volatile int *)0x40005444 = (*(volatile int *)0x40005444 & (~(0x3 << 6))) |
+                                          (0x2 << 6);
+
+            /* Wait for VCOREB to be ready */
+            while (!(MXC_SIMO->buck_out_ready & MXC_F_SIMO_BUCK_OUT_READY_BUCKOUTRDYB)) {}
+
+            /* Lower VregB to reduce power consumption */
+            MXC_SIMO_SetVregO_B(900);
+
+            /* Move VCORE switch to VCOREB (< VCOREA) */
+            MXC_MCR->ctrl = (MXC_MCR->ctrl & ~(MXC_F_MCR_CTRL_VDDCSW)) |
+                            (0x2 << MXC_F_MCR_CTRL_VDDCSW_POS);
+
+            /* Wait for VCOREA ready.  Should be ready already */
+            while (!(MXC_SIMO->buck_out_ready & MXC_F_SIMO_BUCK_OUT_READY_BUCKOUTRDYC)) {}
+        }
+    }
+
+    MXC_LP_ClearWakeStatus();
+
+    /* Set SLEEPDEEP bit */
+    MXC_PWRSEQ->lpcn &= ~MXC_F_PWRSEQ_LPCN_BCKGRND;
+    SCB->SCR |= SCB_SCR_SLEEPDEEP_Msk;
+
+    /* Go into Deepsleep mode and wait for an interrupt to wake the processor */
+    __WFI();
+
+    /*void recoverFromDeepSleep(void)*/
+    {
+        /* SIMO soft start workaround on wakeup */
+        /*void MXC_LP_recoverFromDeepSleep(void)*/
+        {
```

Comment:
Not sure if I'm missing something or if I accidentally deleted some files while reverting the clang-format, but there are hanging curly braces in lp_me14.c.

Reply:
Braces should be balanced. Some extra brace pairs are just leftovers from copy-paste a function. Function name commented out, but braces were left to show that function limits. I left extra braces in since they do no harm. Can be removed.

Reply:
Got it. I think it'd be best to remove the curly braces and the unnecessary commented-out code, so it's more readable for users.

Reply:
> Got it. I think it'd be best to remove the curly braces and the unnecessary commented-out code, so it's more readable for users.

Done.